Merge feature/friendships v2

sebGilR opened the pull request:
For this milestone friendships have been improved:
- When a friendship request is accepted, the existing friendship is confirmed and a new confirmed friendship is created for the user accepting the invitation, thus making it a reciprocate friendship.

Commit:
Create friendship model and table

Commit:
User friendship methods created

Commit:
Add accept method to confirm friendship

Commit:
Start spec files

enelesmai committed:
added test unit for user model

Commit:
Create friendship controller with index, create and update
friendship index view file created

Commit:
Friendship request buttons added to views
Used helper method to define if friendship can be requested

enelesmai committed:
added feature test case for friendships

enelesmai committed:
added flash notices to request answer

enelesmai committed:
added fixes after rubocop

Commit:
remove unrecognized cops

Commit:
Add link to friendships in nav and create reciprocate friendship

maelfosso reviewed (changes requested):
# Changes required :red_circle: 

Hello @sebGilR @enelesmai 

Nice introduction to the milestone.
There are some issues that need to be mentioned

1. Posts timeline includes posts written by logged-in user and their friends ([screenshot (https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

```ruby
# User model

def friends_and_own_posts
  Post.where(user: (self.friends + self))
  # This will produce SQL query with IN. Something like: select * from posts where user_id IN (1,45,874,43);
end
```

Except that the friendship is OK.

Kindly make the suggested changes and submit a new code review request
Thanks

Mael FOSSO

Commit:
Fix tests and removed unnecessary

Commit:
Fix timeline to show only friends and own posts
implemented inverse friendships
add notification to menu with frienship request count

Commit:
Fix rubocop raised errors

sebGilR commented:
Thank you for the provided feedback. Changes requested have been implemented.

sebGilR commented:
Merged by mistake. Changes were implemented both here and the previous milestone so it merged both at the same time when merging the previous one.

suretrust reviewed (commented):
Hi team, 👋 

### Great job implementing the suggested changes. 💯 Let's get this merged! 👍 

_Good luck with your future milestones and happy coding!_ 🎉
